web/src/message_events.js

```diff
@@ -884,8 +884,31 @@ export function update_messages(events) {
         // edited. We should replace any_message_content_edited with
         // passing two sets to rerender_messages; the set of all that
         // are changed, and the set with content changes.
-        for (const list of message_lists.all_rendered_message_lists()) {
-            list.view.rerender_messages(messages_to_rerender, any_message_content_edited);
+
+        for (const msg_list of message_lists.all_rendered_message_lists()) {
+            if (any_message_content_edited && !msg_list.data.filter.can_apply_locally()) {
+                // We need to live update any search based views since the
+                // new content change will affect the message we need to render.
+                message_events_util.maybe_add_narrowed_messages(
+                    messages_to_rerender,
+                    msg_list,
+                    message_util.add_new_messages,
+                );
```

Comment:
Does this need to consider the possible of the edit removing the message from the view?

Reply:
Yup! Added a commit for that which I tested locally by editing a message to remove it from the search.